### ISSUE TITLE
NX-OS does not parse 'show cdp neighbors' output.

### ISSUE THREAD

**Issue by theandrelima** [CiscoTestAutomation/genieparser#563]
_**TL;DR:** NX-OS parser for 'show cdp neighbors' seems to be broken, but the IOS/IOS-XE parser for the same command works fine for the same given NX-OS output._  

Versions tested (all impacted): 21.3, 21.6 and 21.9

Trying to parse the following 'show cdp neighbors' output with **genie.libs.parser.nxos.show_cdp.ShowCdpNeighbors** throws an error:

```
device83# show cdp neighbors
Capability Codes: R - Router, T - Trans-Bridge, B - Source-Route-Bridge
                  S - Switch, H - Host, I - IGMP, r - Repeater,
                  V - VoIP-Phone, D - Remotely-Managed-Device,
                  s - Supports-STP-Dispute


Device-ID             Local Intrfce Hldtme Capability  Platform         Port ID
device01.company.com
                    mgmt0          122    R S I     C6807-XL           Gig2/7/14     
device01.company.com
                    Eth1/47        149    R S I     C6807-XL           Ten1/1/30     
device01.company.com
                    Eth1/48        162    R S I     C6807-XL           Ten2/1/30     
device84.company.com
                    Eth1/49        125    R S I s   N5K-C56128P        Eth1/49       
device84.company.com
                    Eth1/50        125    R S I s   N5K-C56128P        Eth1/50       
device81.company.com
                    Eth1/51        127    R S I s   N5K-C56128P        Eth1/51       
device82.company.com
                    Eth1/52        168    R S I s   N5K-C56128P        Eth1/52       
device-00X        Eth103/1/1     125    H         -VM-SPID           eth0          
device-00YY      Eth104/1/37    125    H         ISE-VM-K9          eth0 
```

The error I get with NXOS parser is:
`Missing keys: [['cdp', 'index', 8, 'port_id'], ['cdp', 'index', 9, 'port_id']]`

However, if I try to parse this exact same output with the IOS/IOS-XE parser (_genie.libs.parser.ios.show_cdp.ShowCdpNeighbors_) it works perfectly fine. 

'show version' output snippet just in case:
```
Cisco Nexus Operating System (NX-OS) Software
(...)

Software
  BIOS:      version 3.7.0
  Power Sequencer Firmware: 
             Module 1: v1.0
             Module 1: v5.0
  Fabric Power Sequencer Firmware: Module 1: version v3.0
  Microcontroller Firmware:        version v0.0.0.45
  QSFP Microcontroller Firmware:   
             Module 1: v2.0.0.0
  SFP Microcontroller Firmware:   
             Module 1: v1.1.0.0
             Module 2: v1.16.0.0
  CXP Microcontroller Firmware:   
             Module not detected
  kickstart: version 7.1(4)N1(1)
  system:    version 7.1(4)N1(1)
(...)
```

**Comment by ThomasJRyan** [comment]
Hi there,

This isn't all that odd actually. IOS/IOS-XE and NXOS operating systems differ on quite a few things, CLI output included. It isn't all that strange that your output would work on one OS and not the other. If you were running the command on NXOS and the error was still occurring that would be an issue, but if you're copy/pasting from one OS to the other than this is pretty expected

**Comment by theandrelima** [comment]
Hi @ThomasJRyan 
I don't think you understood the problem.

The output I provided was taken from an NX-OS device and the NX-OS parser IS NOT working on it.
Surprisingly, the IOS/IOS-XE parser is working for it.


Replying some of your comments individually to make it all clearer.

> IOS/IOS-XE and NXOS operating systems differ on quite a few things, CLI output included. 

Ok, absolutely agreed. And that's exactly the point of this issue.

> It isn't all that strange that your output would work on one OS and not the other. 
Well, it is strange when the output is from NX-OS and the NX-OS parser doesn't work for it, whereas the same output (again, take from NX-OS) does work well with an IOS/IOS-XE parser

> If you were running the command on NXOS and the error was still occurring that would be an issue

I'm glad you agree with that, because that is exactly what is happening: trying to parse an NX-OS output with the NX-OS parser is failing. 

> ... but if you're copy/pasting from one OS to the other than this is pretty expected
well, I am doing that, yes... but as said, surprisingly this is what is working! If I take the NX-OS output and try to parse that with IOS/IOS-XE, the parsing happens just fine.


Can we please reopen this?
